src/platform/packages/shared/kbn-esql-ast/src/commands_registry/complete_items.ts

```diff
@@ -215,16 +205,34 @@ export const getCommandAutocompleteDefinitions = (commands: string[]): ISuggesti
 
 export const getDateHistogramCompletionItem: (histogramBarTarget?: number) => ISuggestionItem = (
   histogramBarTarget: number = 50
-) => ({
-  label: i18n.translate('kbn-esql-ast.esql.autocomplete.addDateHistogram', {
-    defaultMessage: 'Add date histogram',
-  }),
-  text: `BUCKET($0, ${histogramBarTarget}, ${TIME_SYSTEM_PARAMS.join(', ')})`,
-  asSnippet: true,
-  kind: 'Issue',
-  detail: i18n.translate('kbn-esql-ast.esql.autocomplete.addDateHistogramDetail', {
-    defaultMessage: 'Add date histogram using bucket()',
-  }),
-  sortText: '1',
-  command: TRIGGER_SUGGESTION_COMMAND,
-});
+) =>
+  withTriggerSuggestionDialog({
+    label: i18n.translate('kbn-esql-ast.esql.autocomplete.addDateHistogram', {
+      defaultMessage: 'Add date histogram',
+    }),
+    text: `BUCKET($0, ${histogramBarTarget}, ${TIME_SYSTEM_PARAMS.join(', ')})`,
+    asSnippet: true,
+    kind: 'Issue',
+    detail: i18n.translate('kbn-esql-ast.esql.autocomplete.addDateHistogramDetail', {
+      defaultMessage: 'Add date histogram using bucket()',
+    }),
+    sortText: '1',
+  });
+
+/**
+ * Given a suggestion item, decorates it with editor.action.triggerSuggest
+ * that triggers the autocomplete dialog again after accepting the suggestion.
+ *
+ * If the suggestion item already has a custom command, it will preserve it.
+ */
+export function withTriggerSuggestionDialog(suggestionItem: ISuggestionItem): ISuggestionItem {
```

Comment:
```suggestion
export function withAutoSuggest(suggestionItem: ISuggestionItem): ISuggestionItem {
```

I think this is a shorter and more descriptive name wdyt?